### PR TITLE
[tests-only] Add large file upload e2e test

### PR DESCRIPTION
**Pull request by saw-jan**
## Description
Added e2e tests for uploading large file
- pause
- cancel
- resume

## Related Issue
Part of https://github.com/owncloud/ocis/issues/5013

## How Has This Been Tested?
- test environment:

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:
- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

**Comment by ownclouders** [comment]
Results for <strong>e2e-tests oCIS</strong> https://drone.owncloud.com/owncloud/web/35953/12/1
<details><summary>:boom: To see the trace, please open the link in the console ...</summary>

<p>


#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/35953/tracing/upload-resources-in-personal-space-alice-2023-5-24-04-48-14.zip 


</p></details>
:boom: The e2e-ocis tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>e2e-tests oC10</strong> https://drone.owncloud.com/owncloud/web/35971/11/1
<details><summary>:boom: To see the trace, please open the link in the console ...</summary>

<p>


#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/35971/tracing/upload-multiple-small-files-alice-2023-5-24-09-15-48.zip 


</p></details>
:boom: The e2e-oc10 tests pipeline failed. The build has been cancelled.

File: tests/e2e/cucumber/features/smoke/uploadResumable.ocis.feature [hunk 1]
```diff
@@ -0,0 +1,25 @@
+Feature: Upload
+  As a user
+  I want to upload resources
+  So that I can store them in owncloud
+
+  Scenario: Upload resources in personal space
```
[the remark this answers is not in the document]
Comment (saw-jan) [reply tests/e2e/cucumber/features/smoke/uploadResumable.ocis.feature:6]: I forgot to reword after copy-paste

File: tests/e2e/cucumber/features/smoke/uploadResumable.ocis.feature [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    Given "Admin" creates following user using API
+      | id    |
+      | Alice |
+    And the user creates a file "largefile.txt" with size "1GB" in the temp upload directory
```
[the remark this answers is not in the document]
Comment (saw-jan) [reply tests/e2e/cucumber/features/smoke/uploadResumable.ocis.feature:10]: > If I want to upload 500MB will it work?

yes, that should work

> or maybe give selection for user? "([^"]*)" (GB|MB|KB|B)

Great! I will change it to this

File: tests/e2e/cucumber/steps/ui/resources.ts [hunk 3]
```diff
@@ -5,7 +5,14 @@ import { objects } from '../../../support'
...
+import * as tempFs from '../../../support/utils/runtimeFs'
+
+When(
+  'the user creates a file {string} with size {string} in the temp upload directory',
+  function (this: World, fileName: string, fileSize: string): Promise<void> {
+    return tempFs.createFileWithSize(fileName, tempFs.getBytes(fileSize))
+  }
+)
 
 When(
```
[the remark this answers is not in the document]
Comment (saw-jan) [reply tests/e2e/cucumber/steps/ui/resources.ts:17]: Done in this PR. should not be much for this PR

File: tests/e2e/support/objects/app-files/resource/actions.ts [hunk 4]
```diff
@@ -291,54 +297,55 @@ export const uploadResource = async (args: uploadResourceArgs): Promise<void> =>
...
-    await waitForResources({
-      page: page,
-      names: resources.map((file) => path.basename(file.name))
-    })
   }
+
+  await Promise.all([
+    page.waitForResponse((resp) => resp.status() === 201 && resp.request().method() === 'POST'),
+    action
+  ])
```
Comment (saw-jan) [inline tests/e2e/support/objects/app-files/resource/actions.ts:322]: this makes other flaky

File: tests/e2e/cucumber/steps/fs.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import { Given } from '@cucumber/cucumber'
+import { World } from '../environment'
+import * as tempFs from '../../support/utils/runtimeFs'
+
+Given(
+  'the user creates a file {string} of {string} size in the temp upload directory',
+  function (this: World, fileName: string, fileSize: string): Promise<void> {
+    return tempFs.createFileWithSize(fileName, tempFs.getBytes(fileSize))
+  }
+)
```
[the remark this answers is not in the document]
Comment (saw-jan) [reply tests/e2e/cucumber/steps/fs.ts:10]: yeah, should come up with a better name

File: tests/e2e/support/utils/runtimeFs.ts [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+
+// max file creation size is 10GB
+export const MAX_FILE_SIZE = Math.pow(1024, 3) * 10
+
+export const getBytes = (fileSize: string): number => {
+  fileSize = fileSize.replace(/\s/g, '').toLowerCase()
+  const size = parseFloat(fileSize.match(/([\d.]+)/)[0])
+  const type = fileSize.match(/[kKmMgGbB]{1,2}/)[0]
+
+  let sizeInbytes = size
```
Comment (amrita-shrestha) [inline tests/e2e/support/utils/runtimeFs.ts:13]: for this file name too

Comment (saw-jan) [reply tests/e2e/support/utils/runtimeFs.ts:13]: I want to relate it with the node `fs` library. helper for files/folders
Suggest me some names.

**Review by ScharfViktor — APPROVED** [review]
LGTM

File: tests/e2e/cucumber/features/smoke/uploadResumable.ocis.feature [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+      | largefile.txt |
+    And "Alice" pauses the file upload
+    And "Alice" cancels the file upload
+    # issue: https://github.com/owncloud/web/issues/9080
+    # Todo: uncomment when the issue is fixed
+    # And "Alice" starts uploading the following large resources from the temp upload directory
+    #   | resource      |
+    #   | largefile.txt |
+    # And "Alice" pauses the file upload
+    # And "Alice" resumes the file upload
```
Comment (SwikritiT) [inline tests/e2e/cucumber/features/smoke/uploadResumable.ocis.feature:24]: Should there also be a then step to check that the resource is actually uploaded?

**Review by SwikritiT — APPROVED** [review]
LGTM

**Pull request merged**